File: samples/cpp/tutorial_code/introduction/display_image/display_image.cpp
//! [includes]
#include <opencv2/core.hpp>
#include <opencv2/imgcodecs.hpp>
#include <opencv2/highgui.hpp>

#include <iostream>

using namespace cv;
//! [includes]

int main()
{
    //! [imread]
    String image_path = samples::findFile("starry_night.jpg");
    Mat img = imread(image_path, IMREAD_COLOR);
    //! [imread]

    //! [is_empty]
    if(img.empty())
    {
        std::cout << "Could not read the image: " << image_path << std::endl;
        return 1;
    }
    //! [is_empty]

    //! [imshow]
    imshow("Display window", img);
    int k = waitKey(0); // Wait for a keystroke in the window
    //! [imshow]

    //! [imwrite]
    if(k == 's')
    {
        imwrite("starry_night.png", img);
    }
    //! [imwrite]

    return 0;
}

File: samples/python/tutorial_code/introduction/display_image/display_image.py
## [imports]
import cv2 as cv
import sys
## [imports]
## [imread]
tm = cv.TickMeter()
tm.start()
img = cv.imread(cv.samples.findFile("animated-webp-supported.webp"))
tm.stop()
print(tm.getTimeSec())
## [imread]
## [empty]
if img is None:
    sys.exit("Could not read the image.")
## [empty]
## [imshow]
cv.imshow("Display window", img)
k = cv.waitKey(0)
## [imshow]
## [imsave]
if k == ord("s"):
    cv.imwrite("aanimated-webp-supported.webp", img)
## [imsave]
